src/herder/HerderImpl.cpp

```diff
@@ -1031,7 +1031,8 @@ HerderImpl::sendSCPStateToPeer(uint32 ledgerSeq, PeerPtr peer)
         minSeq = maxSeq = ledgerSeq;
     }
 
-    for (uint32 seq = minSeq; seq <= maxSeq; seq++)
+    // use uint64_t for seq to prevent overflows
```

Comment:
GetSCPState message was sent with SCPLedgerSeq equal to UINT32_MAX.

Now we have stacktrace:

void Peer::recvGetSCPState(StellarMessage const& msg)
void HerderImpl::sendSCPStateToPeer(uint32 ledgerSeq, PeerPtr peer) (with ledgerSeq = UINT32_MAX)

This makes:

minSeq = maxSeq = UINT32_MAX; in HerderImpl::sendSCPStateToPeer

and the loops is:

for (uint32 seq = UINT32_MAX; seq <= UINT32_MAX; seq++)
{
}

In first iteration seq = UINT32_MAX, which satisfies UINT32_MAX <= UINT32_MAX
In second it overslows to 0, which satisfies 0 <= UINT32_MAX
Then it goes slowly back to seq = UINT32_MAX and overflows to 0 again